src/migrations/1659435823368_mempool_indexes.ts

```diff
@@ -0,0 +1,6 @@
+import { MigrationBuilder, ColumnDefinitions } from 'node-pg-migrate';
+
+export async function up(pgm: MigrationBuilder): Promise<void> {
+  pgm.createIndex('mempool_txs', [{ name: 'fee_rate', sort: 'DESC' }]);
+  pgm.createIndex('mempool_txs', 'pruned');
```

Comment:
I think this could work great here:
```suggestion
  pgm.createIndex('mempool_txs', 'type_id', { where: 'pruned = false' });
```
What I understand is that having an index on boolean columns is not great when the proportion between true/false values is pretty much the same. However, in the mempool table, unpruned transactions are usually way less than the pruned ones so having this index would really help the query planner.

Could you try this index and give it a spin to see if it gives you a performance boost?

Reply:
@rafaelcr check out the last commit. I was able to get the query planner to use a single index scan, then some pg stats changed in my local db and I couldn't get it working again. But I think in generally the query planner will use the partial indexes when it's generally faster.